test/nuts/delete/source.nut.ts

```diff
@@ -140,6 +141,22 @@ describe('project delete source NUTs', () => {
     expect(fs.existsSync(pathToClass)).to.be.false;
   });
 
+  it('should source:delete an ApexClass only if a specific test pass', async () => {
+    const { apexName, pathToClass } = createApexClass();
+    const response = execCmd<DeleteSourceJson>(
+      `project:delete:source --json --no-prompt --metadata ApexClass:${apexName} --test-level RunSpecifiedTests --tests GeocodingServiceTest`,
+      {
+        ensureExitCode: 0,
+      }
+    ).jsonOutput?.result;
+
+    expect(response?.runTestsEnabled).to.be.true;
+    expect(ensureArray(response?.details.runTestResult?.failures).length).to.equal(0);
+    expect(ensureArray(response?.details.runTestResult?.successes).length).to.be.greaterThanOrEqual(1);
```

Comment:
I tried using `[testkit. apexTestsToBeRun](https://github.com/salesforcecli/source-testkit/blob/main/src/assertions.ts#L326)` to verify that tests were run but it seems to work only for deploy commands:

https://github.com/salesforcecli/source-testkit/blob/main/src/assertions.ts#L326
https://github.com/salesforcecli/plugin-deploy-retrieve/blob/98954e419773c7c8e67901e28cfaa7b72ae5f7d7/test/nuts/seeds/deploy.metadata.test-level.seed.ts#L55